parser.hh

```diff
@@ -181,6 +181,13 @@ class Input
 		return iterator(this, size());
 	}
 	/**
+	 * Returns a user-meaningful name (typically a filename).
+	 */
+	std::string name() const
```

Comment:
This probably should return a const reference, to avoid copying the string every time that it's called.